pyproject.toml

```diff
@@ -20,6 +20,9 @@ dependencies = [
     "ipykernel",
     "black",
     "pyyaml",
+
+    # Pin minimum versions if we've had problems.
```

Comment:
would it be better to pin this to an exact version? That way we control exactly what is being used in the app.

Reply:
That is possible, but there are downsides.
- Pinning to exact version numbers makes it much harder to even think of integrating this tool into a larger project.
- and makes it harder to do upgrades when they are necessary.
- This is a transitive dependency... even if we had pinned the top level dependencies, it wouldn't necessarily have caught this.
- I have filed #361

This is also partially a cultural question. No single dependency manager has a "lock" on the market in python. I think what I'm doing lines up with common Python practices, but I could be wrong. If we do want to handle this differently, I'd look for alternatives withing the python ecosystem.

Reply:
Ok, I'm basing my thought on what is done in package.json for javascript, we always use exact versions. But if it's a different convention in Python,  that's ok